toolbox/oncall_alert/main.go

```diff
@@ -24,120 +24,84 @@ import (
-	gubernatorURL = "https://k8s-gubernator.appspot.com/build/istio-prow"
-
-	doNotMergeLabel = "PostSubmit Failed/Contact Oncall"
-
-	// Token and password file
-	tokenFileDocker        = "/etc/github/git-token"
-	gmailAppPassFileDocker = "/etc/gmail/gmail-app-pass"
-)
-
 var (
 	gcsClient  *u.GCSClient
 	githubClnt *u.GithubClient
 
-	gcsBucket        = flag.String("bucket", "istio-prow", "Prow artifact GCS bucket name.")
-	interval         = flag.Int("interval", 5, "Check and report interval(minute)")
-	debug            = flag.Bool("debug", false, "Optional to log debug message")
-	owner            = flag.String("owner", "istio", "Github owner or org")
-	tokenFile        = flag.String("github_token", tokenFileDocker, "Path to github token")
-	gmailAppPassFile = flag.String("gmail__app_password", gmailAppPassFileDocker, "Path to gmail application password")
-	protectedRepo    = flag.String("protected_repo", "istio", "Protected repo")
-	protectedBranch  = flag.String("protected_branch", "master", "Protected branch")
-	autoBlocking     = flag.Bool("auto_blocking", true, "Blocking auto-merge if tests are failing")
-	emailSending     = flag.Bool("email_sending", true, "Sending alert email")
-
-	postSubmitJobs []*postSubmitJob
-
-	protectedPostsubmits = []string{"istio-postsubmit", "e2e-suite-rbac-auth", "e2e-suite-rbac-no_auth"}
-	receivers            = []string{oncallMaillist}
-	gmailAppPass         string
-	location             *time.Location
+	jobsWatched       []*jobStatus
+	protectedJobs     = []string{"istio-postsubmit", "e2e-suite-rbac-auth", "e2e-suite-rbac-no_auth"}
+	receivers         = []string{oncallMaillist}
+	gmailAppPass      string
+	location          *time.Location
+	pendingJobTimeout = 60 * time.Minute
 )
 
 func init() {
 	flag.Parse()
-
-	var err error
 	gcsClient = u.NewGCSClient()
-
-	token, err := u.GetAPITokenFromFile(*tokenFile)
-	if err != nil || token == "" {
-		log.Fatalf("Error accessing user supplied token_file: %v\n", err)
+	var err error
+	if *guardMaster {
+		token, err := u.GetAPITokenFromFile(*tokenFile)
+		if err != nil {
+			log.Fatalf("Error accessing user supplied token_file: %v\n", err)
+		}
+		githubClnt = u.NewGithubClient(*owner, token)
 	}
-
-	githubClnt = u.NewGithubClient(*owner, token)
-
-	if gmailAppPass, err = u.GetPasswordFromFile(*gmailAppPassFile); err != nil {
-		log.Fatalf("Error accessing gmail app password: %v", err)
+	// TODO set to true
+	if *emailSending {
+		if gmailAppPass, err = u.GetPasswordFromFile(*gmailAppPassFile); err != nil {
+			log.Fatalf("Error accessing gmail app password: %v", err)
+		}
 	}
-
-	for _, t := range protectedPostsubmits {
-		postSubmitJobs = append(postSubmitJobs,
-			&postSubmitJob{
-				name: t,
-				// init to be true to avoid false negative
-				latestRunPass: true,
-			})
+	// Ensure all reruns commands are issued to Prow
+	if _, err := u.Shell(`gcloud container clusters get-credentials prow \
+		--project=istio-testing --zone=us-west1-a`); err != nil {
```

Comment:
Avoid hardcoding prow cluster info

Reply:
It is necessary to have this check tho. Do you think defining them in flags as default value is a good idea? Or is there some configuration file I could access to get this info at runtime?

Reply:
just define it as a constant at least.

Reply:
Done